serverless-plugin/index.js

```diff
@@ -343,8 +326,6 @@ class ServerlessPlugin {
             );
         }
 
-        this.writeHandlers(handlerOptions);
-
         // Ensure the runtime is set to a sane value for other plugins
         if (service.provider.runtime == HASKELL_RUNTIME) {
```

Comment:
Note that this is how the Haskell functions are detected. This will now likely break for _other_ functions that are using _a_ provided runtime but not the Haskell one.

Reply:
Alright, this makes sense, how would I go fixing this?

Reply:
Change `HASKELL_RUNTIME` to what it was before. Then apart from setting the runtime to `provided` [here](https://github.com/seek-oss/serverless-haskell/blob/master/serverless-plugin/index.js#L279), you can add the layer reference.